__tests__/app.test.js

```diff
@@ -164,6 +164,38 @@ describe("Articles Endpoint", () => {
           });
       });
     });
+    describe("Filtering by topic", () => {
+      test("GET:200 - Responds with an array containing correctly formated article objects with all objects having a topic matching the query parameter", () => {
+        return request(app)
+          .get("/api/articles?topic=mitch")
+          .expect(200)
+          .then(({ body }) => {
+            const articles = body.articles;
+
+            expect(articles.length).not.toBe(0);
+            articles.forEach((article) => {
+              expect(article).toMatchObject({
+                article_id: expect.any(Number),
+                title: expect.any(String),
+                topic: "mitch",
+                author: expect.any(String),
+                created_at: expect.any(String),
+                votes: expect.any(Number),
+                article_img_url: expect.any(String),
+                comment_count: expect.any(Number),
+              });
+            });
+          });
+      });
+      test("GET:404 - Responds with an error when passed a topic not present in our database", () => {
+        return request(app)
+          .get("/api/articles?topic=bad_request")
+          .expect(404)
+          .then(({ body }) => {
+            expect(body.msg).toBe("Articles with topic bad_request not found");
+          });
+      });
```

Comment:
One behaviour untested here:
We have a valid topic with no associated articles (topic=paper). In this case i would expect status 200 responds with an empty array of articles